mailet/api/src/main/java/org/apache/mailet/Mail.java

```diff
@@ -276,9 +321,28 @@ default boolean hasSender() {
      *      or null if there was no such attribute (or if the attribute existed
      *      and its value was null)
      * @since Mailet API v2.1
+     * @deprecated see {@link #setAttribute(Attribute)
      */
+    @Deprecated
     Serializable setAttribute(String name, Serializable object);
 
+    /**
+     * Associates an attribute with the given name and value with this Mail instance.
+     * If an attribute with a given name already exists, it is replaced, and the
+     * previous value is returned.
+     * <p>
+     * Conventionally, attribute names should follow the namespacing guidelines
+     * for Java packages.
+     * The Mailet API specification reserves names matching
+     * <i>org.apache.james.*</i> and <i>org.apache.mailet.*</i>.
+     *
+     * @return the previously existing attribute with the same name,
+     *      or null if there was no such attribute (or if the attribute existed
+     *      and its value was null)
+     * @since Mailet API v3.2
+     */
+    Attribute setAttribute(Attribute attribute);
```

Comment:
Sorry I missed this in my previous reviews, but ain't we supposed to no more design APIs using null?